File: algorithms/day-of-programmer/day_of_programmer.py
def gregorianCalendar(y:int) -> int:
  dd = int()

  if (y % 4 == 0) and (y % 100 != 0):
    dd = 12
  elif (y % 400 == 0):
    dd = 12
  else:
    dd = 13

  return dd

def julianCalendar(y:int) -> int:
  return 12 if (y % 4 == 0) else 13


def dayOfProgrammer(year):
  dd = int()
    
  if not(year == 1918):
    if (year > 1918):
      dd = gregorianCalendar(year)
    else:
      dd = julianCalendar(year)
  else:
    dd = 26
                
  return f"{dd}.09.{year}"

File: algorithms/day-of-programmer/dayOfProgrammer.php
<?php
function gregCalendarLeapYear($y) {
  $dd = 13;

  if ($y % 4 == 0 && $y % 100 != 0):
    $dd = 12;
  elseif ($y % 400 == 0):
    $dd = 12;
  endif;

  return $dd;
}

function julianCalendarLeapYear($y) {
  if ($y % 4 == 0):
    return 12;
  endif;
    
  return 13;
}

function dayOfProgrammer($year) {
  $dd = 26;

  if (!($year == 1918)):
    if ($year > 1918):
      $dd = gregCalendarLeapYear($year);
    else:
      $dd = julianCalendarLeapYear($year);
    endif;
  endif;

  return "$dd.09.$year";
}
?>
